client/packages/common/src/utils/numbers/NumUtils.ts

```diff
@@ -19,11 +19,25 @@ export const NumUtils = {
 
     return constrain(parsed, min, max);
   },
+  /**
+   * Round a value to a given precision.
+   *
+   * For example, round(10.232, 2) gives 10.23
+   */
   round: (value: number, dp = 0): number => {
     if (dp === Infinity) return value;
     const multiplier = 10 ** dp;
     return Math.round(value * multiplier) / multiplier;
   },
+  /**
+   * Multiplies two float numbers avoiding "float artefacts".
+   * For example: 110.4 * 29 = 3201.6000000000004
+   * while floatMultiply(110.4, 29) = 3201.6
+   */
+  floatMultiply: (left: number, right: number): number => {
+    // Use a hacky(?) correction factor of 10:
+    return (10 * left * right) / 10;
```

Comment:
`(10 * 1.001 * 1000) / 10;`
`1000.9999999999998`

I think we should maybe create some test cases here, and define the parameters/ranges of data we expect this to work with?
E.g. Do we know our users never use more than 2 decimal places for example?


`(100 * 1.001 * 1000) / 100;`
`1001`

Potentially this is better?

I think there's no 100% reliable way to deal with this issue.

Reply:
thanks good idea, 100 is a better factor! I added some small test as well